perceval/algorithm/abstract_algorithm.py

```diff
@@ -44,7 +44,10 @@ def __init__(self, processor: AProcessor, **kwargs):
         if not self._check_compatibility():
             raise RuntimeError("Processor and algorithm are not compatible")
         self.default_job_name = None
-        self._max_shots = kwargs.get(self._MAX_SHOTS_NAMED_PARAM)
+        try:
+            self._max_shots = int(kwargs.get(self._MAX_SHOTS_NAMED_PARAM))
+        except:
+            self._max_shots = None
```

Comment:
Here, you should raise an exception. We require users to feed a shots threshold to algorithms, else they could burn their shots (credits) without knowing about it.

Reply:
This is already checked at the start of this method, for remote jobs only.

Many TU are designed without max_shots